psga/common/primer_autodetection.nf

```diff
@@ -59,34 +59,59 @@ process primer_autodetection {
 
   # extract 1) the primer with the highest number of reads and 2) the number of reads
-  detected_primer="$(awk 'BEGIN{nreads=0;primer="unknown"}{if(NR>1 && $4>nreads) {nreads=$4; primer=$1}} END {print primer,nreads}' ${sample_id}_primer_detection.tsv)"
+  detected_primer="$(awk 'BEGIN{nreads=0;primer="none"}{if(NR>1 && $4>nreads) {nreads=$4; primer=$1}} END {print primer,nreads}' ${sample_id}_primer_detection.tsv)"
   primer="$(echo "${detected_primer}" | awk '{print $1}')"
   primer_nreads="$(echo "${detected_primer}" | awk '{print $2}')"
 
   # store the data for the selected primer. Use CSV for convenience here
   awk -v name="${primer}" 'NR==1 || $1==name {$1=$1; print}' OFS=, ${sample_id}_primer_detection.tsv > ${sample_id}_primer_data.csv.tmp
-  # store the primer scheme name/version
-  echo "${primer}" > ${sample_id}_primer.txt
 
   # add additional columns
   qc_col="primer_qc"
   autodetect_col="primer_autodetected"
+  orig_kit_col="input_kit"
   if [[ "${kit}" == "unknown" ]]; then
     autodetect_val="True"
     if [[ ${primer_nreads} -gt 0 ]]; then
-      qc_val="Pass"
+      qc_val="PASS"
     else
-      qc_val="Fail"
+      qc_val="FAIL"
+    fi
+  elif [[ "${kit}" == "none" ]]; then
+    autodetect_val="True"
+    if [[ ${primer_nreads} -gt 0 ]]; then
+      qc_val="FAIL"
+    else
+      qc_val="PASS"
     fi
   else
     autodetect_val="False"
     if [[ "${kit}" == "${primer}" ]]; then
-      qc_val="Pass"
+      qc_val="PASS"
     else
-      qc_val="Fail"
+      qc_val="FAIL"
     fi
   fi
-  awk -v qc="${qc_col}" -v ad="${autodetect_col}" -v qc_val="${qc_val}" -v ad_val="${autodetect_val}" 'BEGIN{FS=OFS=","}{print $0,(NR==1 ? qc OFS ad : qc_val OFS ad_val)}' ${sample_id}_primer_data.csv.tmp > ${sample_id}_primer_data.csv
+  awk -v qc="${qc_col}" -v ad="${autodetect_col}" -v kit="${orig_kit_col}" -v qc_val="${qc_val}" -v ad_val="${autodetect_val}" -v kit_val="${kit}" \
+    'BEGIN{FS=OFS=","} {print $0,(NR==1 ? qc OFS ad OFS kit : qc_val OFS ad_val OFS kit_val)}' \
+    ${sample_id}_primer_data.csv.tmp > ${sample_id}_primer_data.csv
 
+  # store the primer scheme name/version
+  echo "${primer}" > ${sample_id}_primer_${qc_val}.txt
+
+  # add a default line if no primer was autodetected
+  if [[ "${primer}" == "none" ]]; then
+      echo "${primer},,,0,0,0,0,0,0,${qc_val},${autodetect_val},${kit}" >> ${sample_id}_primer_data.csv
+
+      # WARNING: this is a hack
+      # certain samples could not have primers at all (e.g. bam files in which adapters and primers were trimmed in the past)
+      # ncov and artic expect a primer scheme and version as input parameters, though.
+      # here we mock ncov by passing a primer scheme/version to make it run.
+      # This won't have consequences because none of the sequences in this primer was found in the sample,
+      # so they won't be trimmed
+      if [[ "${kit}" == "none" ]]; then
+          echo "ARTIC_V3" > ${sample_id}_primer_${qc_val}.txt
+      fi
```

Comment:
I don't like this, but I don't think there are quick alternatives as `artic` requires the primer name and version.

We haven't ever really considered this scenario and the approach has been to give it a primer scheme no matter what.